### PR TITLE
fix: 🤔 Vue: Using registerIconLibrary with wrappers lead to bigger bundle size

### LINKED ISSUE #553: fix: 🤔 Vue: Using registerIconLibrary with wrappers lead to bigger bundle size
# Checklist (for the issue creator)

- [x] I have verified the bug exists in the latest version of `@synergy-design-system/components` (e.g. via `pnpm install @synergy-design-system/components`).
- [x] I can provide a simple, reproducible example of the issue (e.g. via code-sandbox or sample code, ...).
- [x] I would be willing to provide a PR for this issue.

# Version

The bug occurred with the following version of the component library: 2.4.1

# Component

The issue arises when using the following component(s): Icons

# Environment

The issue occurred with the following operating system(s): Node (LTS), Vite (latest)
The issue occurred in the following browser(s): All

# Description

When using `registerIconLibrary` to use a different icon-set, it leads to:
-  warnings when using lit setups. This is caused by the fact that Wrappers (e.g. for Vue) will load their own version of the components package when importing.
- treeshaking not possible
- components and lit is packaged 2 times

Simple Example for a Vue.js APP:

```ts
// main.tsx, loads vue app
// Add this line to enable the light theme for your application
import "@synergy-design-system/tokens/themes/light.css";
import "@synergy-design-system/components/index.css";

// Optional: Import the styles package
import "@synergy-design-system/styles/index.css";

import "./assets/main.css";

import { createApp } from "vue";

// 👇 Leads to the multiple version loaded warning
import { registerIconLibrary } from "@synergy-design-system/components";

// 👇 Does not work at all anymore as its its own scope,
// but gets rid of the error message :(
import { registerIconLibrary } from '@synergy-design-system/components/utilities/icon-library.js';

// @todo: DOCUMENT THAT THIS IMPORT HAS TO HAPPEN AFTER ICON LIB!
import App from "./App.vue";

registerIconLibrary("material-symbols", {
  resolver: (name) => {
    return `/material-symbols/${name}.svg`;
  },
  mutator: (svg) => {
    return svg.setAttribute("fill", "currentColor");
  },
});

createApp(App).mount("#app");
```

# Acceptance criteria

- [x] All DoD items successfully completed
- [x] Make sure to provide registerIconLibrary in a way that does not load lit twice (e.g. reexport from the wrapper package?).
- [x] Documentation is adjusted and point out that order matters (known issues)

### PR DESCRIPTION
**Pull request by schilchSICKAG**
# Pull Request

## 📖 Description

This PR fixes an issue for multiple versions of synergy being loaded when importing `registerIconLibrary` from `@synergy-design-system/components`. It does this by adjusting the build step to create bundled JavaScript files from the original vue sfcs.

### 🎫 Issues

Closes #553 

## 👩‍💻 Reviewer Notes

Please have a look at the changed build step. I have opted to add a `_build` script to the `@synergy-design-system/vue` package for this to work. This also means the `vue-tsc` dependency in `@synergy-design-system/components` could be removed.

I tried using rollup directly, however this did not work as intended because of various problems with bundled peerDependencies. Therefore, `vite` was used as `rollup-plugin-vue` is deprecated for this.

The `vite` compilerOptions can now also be removed to get rid of the annoying custom-element warnings.

## 📑 Test Plan

- Check out this repository
- Install all dependencies
- Start the vue demo project
- Add `registerIconLibrary` as described in the code sample in #553
- There should no duplicate loading message.

## ✅ DoD

- [x] I have tested my changes manually.
- [x] I have added automatic tests for my changes (unit- and visual regression tests).
- [x] I have updated the project documentation to reflect my changes (e.g. CHANGELOG, Storybook, ...).
- [ ] ~~I have added documentation to the DaVinci migration guide (if need be)~~
- [x] I have made sure to follow the projects coding and contribution guides.
- [x] I have made sure that the bundle size has changed appropriately.
- [x] I have validated that there are no accessibility errors.
- [ ] ~~I have used design tokens instead of fix css values~~

### CODE REVIEW

**Comment by schilchSICKAG** [comment]
This also fixes the problems when using vitest 🔥 with synergy-design-system/vue. I have adjusted the known issues file accordingly.

File: packages/vue/vite.config.ts
Comment (kirchsuSICKAG) [inline packages/vue/vite.config.ts:1]: Just one thing I noticed:
There are still the `.vue` files in our package.
Do we want this, or is it even necessary, that we ship them?
image

Comment (schilchSICKAG) [reply packages/vue/vite.config.ts:1]: Yes, I have left them there on purpose for the cases you will still need them. As they are pretty lightweight, I did not opt to remove them.

Comment (kirchsuSICKAG) [reply packages/vue/vite.config.ts:1]: sounds reasonable  👍

**Review by kirchsuSICKAG — APPROVED** [review]

**Pull request merged**